mdm/api/src/main/java/com/gentics/mesh/core/db/Tx.java

```diff
@@ -23,22 +27,32 @@ public interface Tx extends BaseTransaction, DaoCollection, CacheCollection, Sec
 	static ThreadLocal<Tx> threadLocalGraph = new ThreadLocal<>();
 
 	/**
-	 * Set the nested active transaction for the current thread.
+	 * Set the nested active transaction for the current Vert.x context, or thread.
 	 * 
 	 * @param tx
 	 *            Transaction
 	 */
 	static void setActive(Tx tx) {
-		Tx.threadLocalGraph.set(tx);
+		Optional.ofNullable(Vertx.currentContext())
+			.ifPresentOrElse(
+					ctx -> ctx.putLocal("tx", Optional.ofNullable(tx)), 
+					() -> Tx.threadLocalGraph.set(tx)
+			);
 	}
 
 	/**
-	 * Return the current active transaction. A transaction should be the only place where this threadlocal is updated.
+	 * Return the current active transaction. A transaction should be the only place where this context/threadlocal is updated.
 	 * 
 	 * @return Currently active transaction
 	 */
 	static Tx get() {
-		return Tx.threadLocalGraph.get();
+		Optional<Context> maybeVertxContext = Optional.ofNullable(Vertx.currentContext());
+		if (maybeVertxContext.isPresent()) {
+			Context vertxContext = maybeVertxContext.get();
```

Comment:
Don't mix the cases where we have an empty Context, and where we have no Context at all.